serverside_challenge_2/challenge/app/services/electricity_fee_calculator_service.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class ElectricityFeeCalculatorService
+  def self.calculate_fees(contract_ampere, usage_kwh)
+    results = []
+
+    ElectricityPlan
+      .includes(
+        :electricity_provider,
+        :electricity_plan_basic_fees,
+        :electricity_plan_usage_fees
+      )
+      .find_each do |plan|
+      basic_fee = plan.electricity_plan_basic_fees.find_by(ampere: contract_ampere.to_s)&.fee
+      next if basic_fee.nil?
+
+      usage_fee = if usage_kwh.zero?
+                    0
+                  else
+                    calculate_usage_fee(
+                      plan,
+                      usage_kwh
+                    )
+                  end
+
+      total_fee = basic_fee + usage_fee
+      results << { provider_name: plan.electricity_provider.name, plan_name: plan.name, price: total_fee }
+    end
+    results
+  end
+
+  def self.calculate_usage_fee(plan, usage_kwh)
+    usage_fees = plan.electricity_plan_usage_fees.where(min_usage: ...usage_kwh).order(min_usage: :asc)
+
+    # 従量料金設定が1つしかない場合はそのまま掛け算すれば良い
+    return usage_kwh * usage_fees.last.fee if usage_fees.size == 1
+
+    total_fee = 0
+
+    # 従量料金が変動する区間ごとに計算して合計に追加
+    # 現在と次の値のセットのループ
+    # ループの最後の(next_feeがない)場合は処理されない仕様なので、最後の値は別途計算する
+    usage_fees.each_cons(2) do |current_fee, next_fee|
+      next unless usage_kwh > current_fee.min_usage
```

Comment:
```suggestion
      next if usage_kwh <= current_fee.min_usage
```

の方が可読性に優れているように見えますが、unlessを使用した理由はありますか？

Reply:
たしかにifの方が可読性が高いですね、この場合ifにすべきだと思いました。
この形で書いておいてなんですが、個人的にはunlessはrubocopで禁止しても良いかなと思っています。
(条件がbooleanの入った変数1つのみであればアリかなとも思います)